ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/command_submission_service.proto

```diff
@@ -36,7 +36,7 @@ service CommandSubmissionService {
 
 }
 
-// The submitted commands will be processed atomically in a single transaction.
+// The submitted commands will be processed atomically in a single transaction. Each `Command` in `commands` corresponds to an action within that transaction and the order of commands is preserved.
```

Comment:
this is not true -- some commands correspond to multiple "actions". moreover the term "action" is not really well defined here.

i'd change to something like "The commands are executed in the order they were submitted in".

Reply:
It's not true because of the `CreateAndExercise` hack, which, indeed, has broken this correspondence. As for `actions` not being defined, I suggested to @bethaitman to add a kind of glossary to the "Building Applications" docs. I think we should explain what `transactions`, `actions`, `events`, `commands`, `submissions` and `completions` are and how they relate to each other.

I'm not adamant about using `action` here, but I do like relating things back to the ledger model. Ultimately, users need to understand the link between the commands they submit and the resulting transaction, both in terms of ledger model and in terms of event stream. The fact that API and Ledger Model use different terminology and that we have now even broken the 1-1 correspondence between `command` and `event`/`action` is rather unfortunate.

Reply:
@bame-da there was never an intended correspondence between commands and transaction shape. i wouldn't block this pr on a glossary discussion: adding a note about ordering is useful progress without it.